database.go

```diff
@@ -221,6 +221,34 @@ func (m *Measurement) addSeries(s *Series) bool {
 	return true
 }
 
+// removeSeries will remove a series from the measurementIndex. Returns true if already removed
+func (m *Measurement) removeSeries(seriesID uint32) bool {
+	if _, ok := m.seriesByID[seriesID]; !ok {
+		return true
+	}
+	s := m.seriesByID[seriesID]
```

Comment:
Can these two lookups of `seriesByID` not be combined? I think it would be more pleasing if there were.